src/PackageVersions/Installer.php

```diff
@@ -124,12 +124,11 @@ private static function writeVersionClassToFile(
         Config $composerConfig,
         RootPackageInterface $rootPackage
     ) {
-        file_put_contents(
-            self::locateRootPackageInstallPath($composerConfig, $rootPackage)
-            . '/src/PackageVersions/Versions.php',
-            $versionClassSource,
-            0664
-        );
+        $installPath = self::locateRootPackageInstallPath($composerConfig, $rootPackage)
+            . '/src/PackageVersions/Versions.php';
+
+        file_put_contents($installPath, $versionClassSource);
+        chmod($installPath, 0664);
```

Comment:
A default `umask()` will usually be `0022` or such, so it is generally OK to use `file_put_contents()`, but this file is a generated PHP file, and therefore vulnerable to code injection if the `umask()` is incorrect.

Interestingly, specifying the mask on `file_put_contents()` doesn't behave as expected :S